include/openssl/evp.h

```diff
@@ -237,6 +237,14 @@ OPENSSL_EXPORT EVP_PKEY *EVP_parse_private_key(CBS *cbs);
 // success and zero on error.
 OPENSSL_EXPORT int EVP_marshal_private_key(CBB *cbb, const EVP_PKEY *key);
 
+// EVP_marshal_private_key_version marshals |key| as a DER-encoded PrivateKeyInfo
+// structure (RFC 5208) or OneAsymmetricKey (RFC 5958) based on the provided |version|,
+// and appends the result to |cbb|. It returns one on success and zero on error.
+//
+// Ed25519 is the only private key that supports marshaling as a v2 PKCS8 structure.
+// All other private key types will return an error.
+OPENSSL_EXPORT int EVP_marshal_private_key_version(CBB *cbb, const EVP_PKEY *key,
```

Comment:
Create explicit functions for each PKCS8 version?

What if something gets created that is not specified in PKCS8 c.f. https://github.com/awslabs/aws-lc/pull/718/files#diff-1ee2dcc1eaa98a4d7ac8d2571dabcf492f6cadeeedf8edd7c2bd6faeca619920R432-R435?

Reply:
Can you give an example, maybe I just don't know what you mean by `c.f.`? Are you trying to say that the arguments could differ over time which would warrant wanting to have different function signatures per version?

Reply:
Looking at this more, having a separate function all together + a separate pointer on `evp_pkey_asn1_method_st` might make the separation clearer on which algorithms support the v2 encoding.